django/utils/log.py

```diff
@@ -189,3 +191,41 @@ def format(self, record):
 
         record.msg = msg
         return super(ServerFormatter, self).format(record)
+
+
+def log_response(message, *args, **kwargs):
+    """Log errors based on response status.
+
+    Log 5xx responses in error level and 4xx in warning level (unless level
+    is explicitly given as a keyword argument). The Response status_code and
+    the entire request are passed to logger extra parameter. This function
+    also avoid duplicated logs.
+
+    """
+
+    response = kwargs.get('response')
+    request = kwargs.get('request', None)
+    logger = kwargs.get('logger', request_logger)
+    level = kwargs.get('level', None)
+    exc_info = kwargs.get('exc_info', None)
+
+    # Check if the response has already been logged
```

Comment:
Or in some cases (e.g. CSRF) it's because we want to log the response under a particular non-default logger. But even in those cases, we never want to double-log a response.